motionplan/planManager.go

```diff
@@ -81,8 +83,18 @@ func (pm *planManager) PlanSingleWaypoint(ctx context.Context,
 	var goals []spatialmath.Pose
 	var opts []*plannerOptions
 
+	subWaypoints := false
+
 	// linear motion profile has known intermediate points, so solving can be broken up and sped up
 	if profile, ok := motionConfig["motion_profile"]; ok && profile == LinearMotionProfile {
+		subWaypoints = true
+	}
+
+	if len(constraintSpec.GetLinearConstraint()) > 0 {
```

Comment:
Curious as to if/when you see the `LinearMotionProfile` conditional on L89 going away at some point?

Reply:
Perhaps in the future; the scope for this project was that existing `extra` functionality would not be changed

Reply:
Yeah, I thought this was the case. Good call.

Reply:
Do you think that should have been in scope?

Reply:
Might be nice to do a follow-up that cleans up vestiges of the now-duplicated `extra` functionality once we are comfortable with the new approach.